src/query/vacuum.c

```diff
@@ -8131,4 +8103,15 @@ vacuum_data::set_last_blockid (VACUUM_LOG_BLOCKID blockid)
 
   m_last_blockid = blockid;
 }
+
+static void
+vacuum_init_data_page_with_last_blockid (THREAD_ENTRY * thread_p, VACUUM_DATA_PAGE * data_page,
+                                         VACUUM_LOG_BLOCKID blockid)
+{
+  vacuum_data_initialize_new_page (thread_p, data_page);
+  data_page->data->blockid = blockid;
+  log_append_redo_data2 (thread_p, RVVAC_DATA_INIT_NEW_PAGE, NULL, (PAGE_PTR) data_page, 0,
```

Comment:
Better to use blockid instead data_page->data->blockid

Reply:
done